src/MacroTools/Extensions/RectangleExtensions.cs

```diff
@@ -24,24 +24,26 @@ public static void AddSound(this Rectangle region, sound soundHandle)
     }
 
     /// <summary>
-    /// Prepares neutral passive units within the specified <paramref name="rectangle"/> according to
+    /// Prepares units belonging to <paramref name="owningPlayer"/> within the specified <paramref name="rectangle"/> according to
     /// the provided <see cref="RescuePreparationMode"/>.
     /// </summary>
     /// <param name="rectangle">The rectangle in which to prepare units.</param>
     /// <param name="rescuePreparationMode">Determines how units are prepared.</param>
     /// <param name="filter">A function that is applied to each unit found in <paramref name="rectangle"/>.
     /// The unit gets rescued if and only if <paramref name="filter"/> returns true.
     /// </param>
-    /// <returns>Returns all neutral passive units not matching the <paramref name="filter"/> in the specified rectangle.</returns>
-    public static List<unit> PrepareUnitsForRescue(this Rectangle rectangle, RescuePreparationMode rescuePreparationMode, Func<unit, bool>? filter = null)
+    /// <param name="owningPlayer">Player the units inside the specified rectangle belong to. The default value is neutral passive.</param>
+    /// <returns>Returns all units belonging to <paramref name="owningPlayer"/> and not matching the <paramref name="filter"/> in the specified rectangle.</returns>
+    public static List<unit> PrepareUnitsForRescue(this Rectangle rectangle, RescuePreparationMode rescuePreparationMode, Func<unit, bool>? filter = null, player? owningPlayer = null)
```

Comment:
I think instead of adding another parameter, we should change the trolls in the map to start owned by Neutral Passive, then use the normal `PrepareUnitsForRescue` and make them hostile in the quest constructor. I'm just worried about giving such a ubiquitous function a signature that's even remotely complex. Open to argument.

Reply:
The downside of your approach for me personally and anyone else who doesn't know how to work with the editor would be that any cases like would require someone else to do editor work just to get cases like these working.
The function does become more complex but the parameter can be safely ignored 99% of the time since it is optional the default value is the same as before and I'd also argue that this is a very natural way of extending the functionality. 
And I don't see these trolls starting as hostile as 'wrong'. This is just a regular creep camp that just happens to be in the way for when a specific quest fails. 
If we were to remove the quest now, everything would work just fine but if we made the camp neutral passive by default we would either have to do additional editor work or move the function call to a different place where it might not fit.

I don't know if I expressed myself well here. Well this is the downside to not being a native english speaker.

Reply:
I agree. I'm especially sold by your point that without this quest, the units should start hostile.

But I think there is a 3rd option that's best. These units aren't really being "prepared" for rescue; they're not being made invulnerable or hidden, nor are they rescued later. They're just being put into a list so they can be iterated across. So I think it's cleanest to not use PrepareUnitsForRescue at all, as in:
`_killUnits = new GroupWrapper().EnumUnitsInRect(rescueRect).Where(x => x.OwningPlayer == Player(PLAYER_NEUTRAL_AGGRESSIVE))`
This would allow us to keep the `PrepareUnitsForRescue` API clean and restrict it to cases of actual rescue.

Reply:
I see your point. This would pollute the api and allow 'abusing' the function, open it up too much.

Reply:
Yeah, pretty much. I think we should add the player parameter if and only if there are neutral hostile units that need to be made invulnerable and/or hidden at game start _and_ which must be rescued later. Right now we don't satisfy that category.